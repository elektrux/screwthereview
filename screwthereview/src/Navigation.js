import React from 'react';
 
import { NavLink } from 'react-router-dom';
 
const Navigation = () => {
    return (
       <div>
          <NavLink to="/">Home</NavLink>
          <NavLink to="/roulette">Roulette</NavLink>
       </div>
    );
}
 
export default Navigation;