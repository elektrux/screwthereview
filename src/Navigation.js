import React from 'react';
import logo from './Images/ScrewtheReviewLogo.png';

function Navigation() {
    return (
        <nav className="navbar navbar-expand-lg navbar-light">
            <a className="navbar-brand mr-l" href="/">
                <img src={logo} alt="logo" style={{ width: '150px' }} />
            </a>

            { /* Collapsable Button on Right */}
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>

            <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav mr-auto" >
                    <li className="nav-item ml-4">
                        <a className="nav-link" href="/">Home</a>
                    </li>
                    <li className="nav-item ml-4" >
                        <a className="nav-link" href="/search">Search</a>
                    </li>
                    <li className="nav-item ml-4">
                        <a className="nav-link" href="/roulette">Roulette</a>
                    </li>
                    <li className="nav-item ml-4">
                        <a className="nav-link" href="/about">About</a>
                    </li>
                    <li className="nav-item ml-4">
                        <a className="nav-link" href="/account">Account</a>
                    </li>
                </ul>
            </div>
        </nav>
    );
}

export default Navigation;
