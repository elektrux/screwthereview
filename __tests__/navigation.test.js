import React from 'react';
import { render, cleanup, screen } from '@testing-library/react'
import Navigation from '../src/Navigation.js';


describe('Nav Bar Component', () => {
  //     // Use .skip annotation to disable tests
  //     test.skip('Navigation bar matches snapshot dom', () => {

  //         // using react-test-renderer
  //         const component = renderer.create(
  //             <Navigation/>
  //         );
  //         let tree = component.toJSON();
  //         expect(tree).toMatchSnapshot();
  //   });

  test("Nav bar contains a button + 5 list entries", () => {
    // using react testing lib
    render(<Navigation />);
    expect(screen.getByRole('button'));
    expect(screen.getAllByRole('listitem').length).toBeGreaterThanOrEqual(4);
    cleanup();
  });
})