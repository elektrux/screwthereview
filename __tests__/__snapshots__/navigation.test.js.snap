// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar Component Navigation bar matches snapshot html and is navigable 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-light"
>
  <a
    className="navbar-brand mr-l"
    href="/"
  >
    <img
      alt="logo"
      src="test-file-stub"
      style={
        Object {
          "width": "150px",
        }
      }
    />
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item ml-4"
      >
        <a
          className="nav-link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="nav-item ml-4"
      >
        <a
          className="nav-link"
          href="/search"
        >
          Search
        </a>
      </li>
      <li
        className="nav-item ml-4"
      >
        <a
          className="nav-link"
          href="/roulette"
        >
          Roulette
        </a>
      </li>
      <li
        className="nav-item ml-4"
      >
        <a
          className="nav-link"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="nav-item ml-4"
      >
        <a
          className="nav-link"
          href="/account"
        >
          Account
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
